src/hooks/useDatabase/index.ts

```diff
@@ -1,34 +1,41 @@
 import { Collection, Application, ModelBase, Status } from '~/@types/common';
+import { DBApplication, DBModelBase } from '~/@types/database';
+import { mapEntry } from './utils';
 
 type DropFirst<T extends unknown[]> = T extends [any, ...infer U] ? U : never;
 
 const FREE_ID_KEY = 'free-id';
 
 const useDatabase = () => {
-  const get = <T extends ModelBase>(
+  const get = <Model extends ModelBase, DBModel extends DBModelBase>(
```

Comment:
Please use uppercase names as name of generic type, like: `MODEL` or `DB_MODEL` - it is easiest to detect that this is a generic type

Reply:
Done